File: benchmarks/sort/python/sort.py
import sys


def main():
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 2_000_000

    state = 42
    arr = [0] * n
    for i in range(n):
        state = (state * 1103515245 + 12345) & 0x7FFFFFFF
        arr[i] = state

    arr.sort()

    assert all(arr[i - 1] <= arr[i] for i in range(1, n)), "self-check failed: array not sorted"
    print(arr[-1])


if __name__ == "__main__":
    main()

File: benchmarks/sort/rust/sort.rs
use std::env;

struct Lcg {
    state: u32,
}

impl Lcg {
    fn next(&mut self) -> i64 {
        self.state = self.state.wrapping_mul(1103515245).wrapping_add(12345) & 0x7fffffff;
        self.state as i64
    }
}

fn main() {
    let n: usize = env::args().nth(1).map(|s| s.parse().unwrap()).unwrap_or(2_000_000);

    let mut lcg = Lcg { state: 42 };
    let mut arr: Vec<i64> = (0..n).map(|_| lcg.next()).collect();

    arr.sort();

    assert!(
        arr.windows(2).all(|w| w[0] <= w[1]),
        "self-check failed: array not sorted"
    );
    println!("{}", arr[n - 1]);
}
